### ISSUE TITLE
release: Do not edit CHANGELOG on every PR, find better way to add changelog items.

### ISSUE THREAD

**Issue by bwplotka** [thanos-io/thanos#3320]
The current way of editing changelog for every PR that is worth mentioning is useful for someone doing the release process. However it's painful due to amount of rebases. Let's find better solution.

AC:
* Minimize work done by release shepherd by allowing author of PR/commit can mention changelog items that would be proposed for the change.
* Reduce number of git conflicts when editing changelog
* Allow reviewing changelog item before submitting PR.

Some relevant discussion on Prometheus side: https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/prometheus-developers/GHR3ZPzbKKQ/7bUCPC3uAgAJ

cc @simonpasquier as you have something potentially nice on Alertmanager. Can you link us to some tooling? :hugs:

**Comment by brancz** [comment]
I was recently contributing to some terraform modules and they used: https://github.com/hashicorp/go-changelog

The process was super smooth. The way it works is that there is a code tag with the "language" changelog in every PR, which is enforced. On releases, this can automatically be extracted with tooling from the PR descriptions. Beyond not creating conflicts all the time, this also allows maintainers of a repo to make edits to the changelog item, helping both maintainers and contributors to ensure things don't fail at the changelog entry.

**Comment by bwplotka** [comment]
Still very much required: See another proof: https://github.com/thanos-io/thanos/pull/3737#pullrequestreview-574178511

**Comment by bwplotka** [comment]
Have some ideas.

What about having some command like `promu bump` that traverse all commits and builds up the CHANGELOG from commits: https://github.com/prometheus/promu/pull/170#issuecomment-786471384 then we will make sure CHANGELOG file is never updated via commit bu we write changelog on commit / PR message.

**Comment by bwplotka** [comment]
This looks less exciting as implementing our own CLI tool, but might be smarter to reuse than write our own 🙃 Looks good: https://github.com/git-chglog/git-chglog

**Comment by squat** [comment]
yes agreed. Let's specialize in the observability space and rely on community for solutions in other areas. This tool looks great!

**Comment by bwplotka** [comment]
Played a little bit and found some good use of git-chglog if they are willing to accept some contributions: https://github.com/git-chglog/git-chglog/issues/101

The problem is that we don't want to just use some commit title convention, we will never get this assured. A simple checkbox list in commit/PR message would work though 🤗

**Comment by bwplotka** [comment]
Maybe commit style is what we could enforce, but I am skeptical 🤗 

Some good discussion: https://twitter.com/JoelOtter/status/1366475156974108673

**Comment by GiedriusS** [comment]
I think we can now simply copy what prometheus/prometheus has.
